projects/plugins/crm/src/automation/interface-step.php

```diff
@@ -36,13 +36,40 @@ public function execute( Data_Type $data );
 	public function get_next_step_id();
 
 	/**
-	 * Set the next step.
+	 * Get the next step if the current one is successful.
 	 *
 	 * @since $$next-version$$
 	 *
-	 * @param int|string|null $step_id The next linked step.
+	 * @return int|string|null The next linked step id.
 	 */
-	public function set_next_step( $step_id );
+	public function get_next_step_true();
+
+	/**
+	 * Set the next step if the current one is successful.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @return int|string|null The next linked step id.
+	 */
+	public function set_next_step_true();
+
+	/**
+	 * Get the next step if the current one is falsy.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @return int|string|null The next linked step id.
+	 */
+	public function get_next_step_false();
+
+	/**
+	 * Set the next step if the current one is falsy.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @return int|string|null The next linked step id.
+	 */
+	public function set_next_step_false();
```

Comment:
Do we need this?

Reply:
Due to the way we've implemented the system, then yes.

Because we use `<something> instanceof Step` then we don't enforce folks to extend our abstract base class (`Base_step`), so 3. party developers could technically avoid the base class right now, so we would force them.

I would personally prefer if we had consistently used `Base_Step` and `is_subclass_of( <something>, Base_Step::class )` so we could assume everything that exists in the abstract base class would be present, but we haven't

Reply:
> we use `<something> instanceof Step`

I'm not seeing that anywhere, to be honest, so I don't see the interface actually being used/needed (other than the fact it's currently referenced in the base class implementation).

To me it's essentially redundant vs. just implementing the base class). I agree that my preference is extending a parent class.

Reply:
I guess it's not as bad as I thought/remembered. I know we have a few methods that type-hints return value as `method(): Step`

Reply:
But I do find it unnecessary as well. I tend to complete move away from interfaces if I have an abstract class because you might as well implement what's in the interface as `abstract` and then add your generic logic as well